providers/coordinator.go

```diff
@@ -207,7 +207,13 @@ func (c *coordinator) RemoveRuntime(runtime *Runtime) {
 	}
 
 	// Shutdown any providers that are not being used anymore
-	if len(c.runtimes) == 0 && len(c.unprocessedRuntimes) == 0 {
+	uprocessedRuntimeWithProviders := false
```

Comment:
can you add a comment here, either in this pr or as a followup, to say why we need this check?

Reply:
done